diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
@@ -163,7 +163,7 @@ class PermissionsService(
     }
 
     val permissions = permissionsList
-      .map { permission -> permission.split(".").last() }
+      .map { permission -> permission.split(".").lastOrNull() ?: permission }
       .filter { permission ->
         permission.isNotEmpty()
       }
