diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
index 0000000..1111111 100644
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
@@ -155,15 +155,15 @@ class PermissionsService(
   private fun requestPermissions(
     permissionsList: Array<String>
   ) {
     if (permissionsList.isEmpty()) {
       return
     }
 
     val permissions = permissionsList
-      .map { permission -> permission.split(".").last() }
+      .map { permission -> permission.split(".").lastOrNull() ?: permission }
       .filter { permission ->
         permission.isNotEmpty()
       }
